arraycontext/impl/pytato/compile.py

```diff
@@ -415,12 +415,23 @@ def _dag_to_transformed_pytato_prg(self, dict_of_named_arrays, *, prg_id=None):
                 prg_id, "pre_generate_loopy", pt_dict_of_named_arrays)
 
         with ProcessLogger(logger, f"generate_loopy for '{prg_id}'"):
+            import pyopencl as cl
+            dev = self.actx.context.devices[0]
```

Comment:
Safer to get it from the queue, which only has one device.

Reply:
Done in [1e54ce4](https://github.com/inducer/arraycontext/pull/189/commits/1e54ce49945684008ab48a1a7f2793fc5590a702)